bagger/src/migration_tools.py

```diff
@@ -255,8 +258,31 @@ def call_dds(delay, filter, total, after):
     for bnumber in bnumber_source:
         print("[")
         url = settings.DDS_GOOBI_NOTIFICATION.format(bnumber)
-        r = requests.get(url)
-        j = r.json()
+
+        try:
+            response = requests.get(url)
+
+            if response.status_code != 200:
+                message = (
+                    f"Got non-200 response from {url}!\n"
+                    f"Status code: {response.status_code}\n"
+                    f"Raw response:\n\n[START]{response.text}[END]\n\n"
+                )
+
+                raise Exception(message)
+
+            j = response.json()
+
+        except ValueError:
+            raise Exception(
+                f"Decoding JSON from  {url} failed for:\n {response.text}\n"
+            )
+
+        except:
```

Comment:
minor: Catch `except Exception` here, not a raw `except`.

The latter will include things like SystemExit and KeyboardInterrupt, which you don't usually want to catch.